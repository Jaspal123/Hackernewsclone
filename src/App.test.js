import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {Search, Button, Table} from './App';
import App from './App';

describe('App', () => {
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('snapshot', () => {
  const component = renderer.create(
    <App />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
})

describe('Search', () => {
  it('renders', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Search>Search</Search>, div)
  })
  test('snapshot', () => {
    const component = renderer.create(
      <Search>Search</Search>
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})


describe('Button', () => {
  it('renders', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button>More</Button>, div)
  })
  test('snapshot', () => {
    const component = renderer.create(
      <Button>More</Button>
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})


describe('Table', () => {
  const props = {
    lists
  }
  it('renders', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button>More</Button>, div)
  })
  test('snapshot', () => {
    const component = renderer.create(
      <Button>More</Button>
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})