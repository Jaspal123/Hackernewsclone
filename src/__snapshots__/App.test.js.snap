// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<div
  className="page"
>
  <div
    className="interactions"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        type="text"
        value="redux"
      />
      <button
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
  <div
    className="table"
  />
  <div
    className="interactions"
  >
    <button
      onClick={[Function]}
      type="submit"
    >
      More
    </button>
  </div>
</div>
`;

exports[`Button snapshot 1`] = `
<button
  type="submit"
>
  More
</button>
`;

exports[`Search snapshot 1`] = `
<form>
  <input
    type="text"
  />
  <button
    type="submit"
  >
    Search
  </button>
</form>
`;
